sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -1091,6 +1111,7 @@ object CodeGenerator extends Logging {
     }
 
     // Then walk the classes to get at the method bytecode.
+    val methodsToByteCodeSize = mutable.ArrayBuffer[(String, Int)]()
```

Comment:
Since we know the number of methods right below (for each generated class, `cf.methodInfos` would give us the number of methods in that class, of which we expect almost all of them to have a `Code` attribute), making pre-sized arrays for each class and then concatenating those arrays into a `Seq` as the result would avoid potential resizing wastes from using a big default-sized `ArrayBuffer`.